src/celeritas/io/ImportUnits.cc

```diff
@@ -0,0 +1,125 @@
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/io/ImportUnits.cc
+//---------------------------------------------------------------------------//
+#include "ImportUnits.hh"
+
+#include <type_traits>
+
+#include "corecel/io/EnumStringMapper.hh"
+#include "corecel/math/Algorithms.hh"
+#include "corecel/math/Quantity.hh"
+#include "celeritas/UnitTypes.hh"
+
+using celeritas::units::visit_unit_system;
+
+namespace celeritas
+{
+namespace
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Get the length scale of a unit system.
+ */
+double length_scale(UnitSystem sys)
+{
+    return visit_unit_system(
+        [](auto traits) {
+            using Unit = typename decltype(traits)::Length;
+            return native_value_from(Quantity<Unit, double>{1});
+        },
+        sys);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Get the time scale of a unit system.
+ */
+double time_scale(UnitSystem sys)
+{
+    return visit_unit_system(
+        [](auto traits) {
+            using Unit = typename decltype(traits)::Time;
+            return native_value_from(Quantity<Unit, double>{1});
+        },
+        sys);
+}
+
+//---------------------------------------------------------------------------//
+}  // namespace
+
+//---------------------------------------------------------------------------//
+/*!
+ * Get a printable label for units.
+ */
+char const* to_cstring(ImportUnits value)
+{
+    static EnumStringMapper<ImportUnits> const to_cstring_impl{
+        "unitless",
+        "MeV",
+        "MeV/len",
+        "len",
+        "1/len",
+        "1/len-MeV",
+        "MeV^2/len",
+        "len^2",
+        "time",
+        "1/len^3",
+    };
+    return to_cstring_impl(value);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Get the native value from a quantity of this type.
+ */
+double native_value_from(UnitSystem sys, ImportUnits q)
+{
+    constexpr double mev = 1;
+    double const len = length_scale(sys);
+
+    switch (q)
+    {
+        case ImportUnits::none:
+            return 1;
+        case ImportUnits::mev:
+            return mev;
+        case ImportUnits::mev_per_len:
+            return mev / len;
+        case ImportUnits::len:
+            return len;
+        case ImportUnits::len_inv:
+            return 1 / len;
+        case ImportUnits::len_mev_inv:
+            return 1 / (len * mev);
+        case ImportUnits::mev_sq_per_len:
+            return ipow<2>(mev) / len;
+        case ImportUnits::len_sq:
+            return ipow<2>(len);
+        case ImportUnits::time:
+            return 1 / ipow<2>(time_scale(sys));
```

Comment:
```suggestion
            return time_scale(sys);
```
Maybe this needs a test?